validator/client/beacon-api/sync_committee.go

```diff
@@ -93,6 +95,37 @@ func (c *beaconApiValidatorClient) getSyncCommitteeContribution(
 	return convertSyncContributionJsonToProto(resp.Data)
 }
 
+func (c *beaconApiValidatorClient) getSyncSubcommitteeIndex(ctx context.Context, in *ethpb.SyncSubcommitteeIndexRequest) (*ethpb.SyncSubcommitteeIndexResponse, error) {
+	validatorIndexResponse, err := c.validatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: in.PublicKey})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get validator index")
+	}
+
+	syncDuties, err := c.dutiesProvider.GetSyncDuties(ctx, slots.ToEpoch(in.Slot), []primitives.ValidatorIndex{validatorIndexResponse.Index})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get sync committee duties")
+	}
+
+	if len(syncDuties) == 0 {
+		return nil, errors.Errorf("no sync committee duty for the given slot %d", in.Slot)
+	}
+
+	// First sync duty is required since we requested sync duties for one validator index.
+	syncDuty := syncDuties[0]
+
+	var indices []primitives.CommitteeIndex
+	for _, idx := range syncDuty.ValidatorSyncCommitteeIndices {
+		syncCommIdx, err := strconv.ParseUint(idx, 10, 64)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse validator sync committee index")
+		}
+
+		indices = append(indices, primitives.CommitteeIndex(syncCommIdx))
+	}
```

Comment:
```suggestion
	indices := make([]primitives.CommitteeIndex, len(syncDuty.ValidatorSyncCommitteeIndices))

	for i, idx := range syncDuty.ValidatorSyncCommitteeIndices {
		syncCommIdx, err := strconv.ParseUint(idx, 10, 64)
		if err != nil {
			return nil, errors.Wrap(err, "failed to parse validator sync committee index")
		}

		indices[i] = primitives.CommitteeIndex(syncCommIdx)
	}
```

Reply:
Hi @nalepae !
Slice initialisation with make is recommended when the length of slice is constant or already known. In our case it length of `indices` depends on length  `syncDuty.ValidatorSyncCommitteeIndices` slice. If `syncDuty.ValidatorSyncCommitteeIndices` slice is empty/nil then it doesn't make sense to pre-allocate indices slice.